client/app/src/services/toast.service.js

```diff
@@ -0,0 +1,97 @@
+(function() {
+  'use strict';
+
+  var fs = require('fs');
+
+  angular.module('arkclient.services')
+    .service('toastService', ['configService', '$mdToast', 'gettextCatalog', ToastService]);
+
+  /**
+   * ToastService
+   * @constructor
+   */
+  function ToastService(configService, $mdToast, gettextCatalog) {
+    self.TypeEnum = {
+      ERROR: 0,
+      SUCCESS: 1,
+      WARN: 2,
+      LOG: 3,
+      DEBUG: 4
+    }
+
+    self.TypeName = [
+      'error',
+      'success',
+      'warn',
+      'log',
+      'debug'
+    ]
+
+    self.logFile = configService.getByGroupAndKey('notice', 'logFile');
+    self.loggingType = configService.getByGroupAndKey('notice', 'level');
+    self.hideDelay = configService.getByGroupAndKey('notice', 'defaultHideDelay');
+    self.fileStream = null;
+
+    function error(message, hideDelay, stopTranslate) {
+      show(message, self.TypeEnum.ERROR, hideDelay, stopTranslate);
+    }
+
+    function success(message, hideDelay, stopTranslate) {
+      show(message, self.TypeEnum.SUCCESS, hideDelay, stopTranslate);
+    }
+
+    function warn(message, hideDelay, stopTranslate) {
+      show(message, self.TypeEnum.WARN, hideDelay, stopTranslate);
+    }
+
+    function log(message, hideDelay, stopTranslate) {
+      show(message, self.TypeEnum.LOG, hideDelay, stopTranslate);
+    }
+
+    function debug(message, hideDelay, stopTranslate) {
+      show(message, self.TypeEnum.DEBUG, hideDelay, stopTranslate);
+    }
+
+    self.show = function(message, type, hideDelay, stopTranslate) {
+      if (type > self.loggingType) {
+        return;
```

Comment:
Maybe we should log an error here before returning?

Reply:
Yeh it may be worth doing this. I will get it to log all messages to the log file, if enabled.